src/main/java/dan200/computercraft/client/render/TileEntityMonitorRenderer.java

```diff
@@ -235,22 +225,19 @@ private static void renderTerminal( @Nonnull MultiBufferSource bufferSource, Pos
                 }
 
                 // Render background geometry
-                bufferSource.getBuffer( RenderTypes.MONITOR );
                 RenderTypes.MONITOR.setupRenderState();
-                backgroundVbo.drawWithShader(
-                    matrix, RenderSystem.getProjectionMatrix(), RenderTypes.getMonitorShader(), backgroundVbo.getIndexCount()
+                vbo.drawWithShader(
+                    matrix, RenderSystem.getProjectionMatrix(), RenderTypes.getMonitorShader(), monitor.backgroundVertexCount, 0
                 );
-                RenderTypes.MONITOR.clearRenderState();
 
                 // Render foreground geometry with glPolygonOffset enabled. Note we don't care about drawing these in
                 // front to back order because it's a cutout type shader that uses discard.
-                bufferSource.getBuffer( RenderTypes.MONITOR_POLYGON_OFFSET );
                 RenderTypes.MONITOR_POLYGON_OFFSET.setupRenderState();
```

Comment:
I wonder if we can get rid of the `bufferSource.getBuffer( RenderType.solid() );` now (or at least in the VBO case). We don't ever touch `MultiBufferSource`. Might save a few draw calls.

I also wonder if it's better doing the polygon offset stuff ourself rather than using render types. In principle I'm not a fan, but it would reduce some work that `setupRenderState` does - worth benchmarking maybe?

Reply:
I tried to removing `bufferSource.getBuffer( RenderType.solid() );` but everything went haywire! Monitor geometry was smearing across the screen. I might be missing a call BufferUploader.reset() somewhere...

`setupRenderState` was showing up in my profiles as the 2nd heaviest method behind the `DirectVertexBuffer`'s draw routines, so I went ahead and setup `glPolygonOffset` directly. It makes a difference in the profile and IMO seems harmless to do directly.

Reply:
Aww, that's sad, but fair enough!